spring-cloud-gateway-server-mvc/src/test/java/org/springframework/cloud/gateway/server/mvc/filter/BeforeFilterFunctionsTests.java

```diff
@@ -127,7 +128,8 @@ void removeRequestParameterWithEncodedRemainParameters() {
 
 		assertThat(result.param("foo")).isEmpty();
 		assertThat(result.param("baz[]")).isPresent().hasValue("qux[]");
-		assertThat(result.uri().toString()).hasToString("http://localhost/path?baz%5B%5D=qux%5B%5D");
```

Comment:
So this test was asserting for the presence of the bug?

Reply:
@ryanjbaxter 
Correct. This test ensures that `quux=corge+` is properly encoded.